doc/en/fixture.rst

```diff
@@ -216,20 +216,189 @@ functions take the role of the *injector* and test functions are the
 .. _`conftest.py`:
 .. _`conftest`:
 
-``conftest.py``: sharing fixture functions
-------------------------------------------
+Fixture availabiility
+---------------------
 
-If during implementing your tests you realize that you
-want to use a fixture function from multiple test files you can move it
-to a ``conftest.py`` file.
-You don't need to import the fixture you want to use in a test, it
-automatically gets discovered by pytest. The discovery of
-fixture functions starts at test classes, then test modules, then
-``conftest.py`` files and finally builtin and third party plugins.
+Fixture availability is determined from the perspective of the test. A fixture
+is only available for tests to request if they are in the scope that fixture is
+defined in. If a fixture is defined inside a class, it can only be requested by
+tests inside that class. But if a fixture is defined inside the global scope of
+the module, than every test in that module, even if it's defined inside a class,
+can request it.
+
+Similarly, a test can also only be affected by an autouse fixture if that test
+is in the same scope that autouse fixture is defined in (see
+:ref:`autouse order`).
+
+A fixture can also request any other fixture, no matter where it's defined, so
+long as the test requesting them can see all fixtures involved.
+
+For example, here's a test file with a fixture (``outer``) that requests a
```

Comment:
The reason for having two classes is to demonstrate that each test class has its own `inner` fixture, and aren't pulling it from the other class' scope. I also didn't want to involve anything about overriding fixtures just yet because that wasn't what this part was about, and there's a section for that already. In order for these tests to pass, both classes must have their own `inner` fixture defined.

Regarding `order`, I pulled this from another example in the docs. I think it's ideal because it demonstrates an actual implication for the tests, whereas prints don't impact the tests. That said, actually _seeing_ it may be beneficial as well. Maybe I should add a note about running it with `--setup-show/plan`?